kythe/typescript/indexer.ts

```diff
@@ -1117,7 +1086,76 @@ class Visitor {
     if (decl.body) {
       this.visit(decl.body);
     } else {
-      if (kFunc) this.emitFact(kFunc, 'complete', 'incomplete');
+      this.emitFact(kFunc, 'complete', 'incomplete');
+    }
+  }
+
+  /**
+   * Visits a function parameters, which can be recursive in the case of
+   * parameters created via bound elements:
+   *    function foo({a, b: {c, d}}, e, f) {}
+   * In this code, a, c, d, e, f are all parameters with increasing parameter
+   * numbers [0, 4]. Ideally the parameter number would be hierarchical, with
+   * `a` having a parameter number of 0.0, but Kythe doesn't support that.
```

Comment:
Why is it idea to have these specific names?  It only needs to be consistent within this module, right?

Reply:
Oops! Removed.